### PR TITLE
add error checking for slr similarity measure in HistogramRegistration

### PR DESCRIPTION
**Pull request by alexis-roche**
This should fix https://github.com/nipy/nipy/issues/273

### CODE REVIEW

File: nipy/algorithms/registration/histogram_registration.py [hunk 1]
```diff
@@ -180,6 +183,12 @@ def subsample(self, spacing=None, npoints=None):
 
     def _set_similarity(self, similarity, renormalize=False, dist=None):
         if similarity in _sms:
+            if similarity == 'slr':
+                if dist == None:
```
Comment (matthew-brett) [inline nipy/algorithms/registration/histogram_registration.py]: Should be `if dist is None`

File: nipy/algorithms/registration/histogram_registration.py [hunk 2]
```diff
@@ -180,6 +183,12 @@ def subsample(self, spacing=None, npoints=None):
...
+                    raise ValueError('slr measure requires a joint intensity distribution model, '\
```
Comment (matthew-brett) [inline nipy/algorithms/registration/histogram_registration.py]: You don't actually need the slash at the end, because of the enclosing parens.

**Comment by matthew-brett** [comment]
Thanks - that's kind of you.  Would you mind adding a test for the raised errors?

**Comment by alexis-roche** [comment]
Yes, sorry, I was not expecting you to react so promptly and was actually preparing a test to be amended to the same commit.

**Comment by alexis-roche** [comment]
Are we good here?

**Comment by matthew-brett** [comment]
Yes, I can imagine you were expecting me to be slower :)

**Comment by matthew-brett** [comment]
Thanks for doing that - looks fine - merging.

**Pull request merged**